### PR TITLE
{yoda,lhapdf,fastnlo-toolkit}: fix and enable strictDeps

### PR DESCRIPTION
**Pull request by FliegendeWurst**
All done in one PR because the packages look very related.

ref. #178468

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/fa/fastnlo-toolkit/package.nix [hunk 1]
```diff
@@ -35,6 +35,11 @@ stdenv.mkDerivation rec {
     ./yoda2_support.patch
   ];
 
+  nativeBuildInputs = [
+    lhapdf # lhapdf-config
+    yoda # yoda-config
```
Comment (veprbl) [inline pkgs/by-name/fa/fastnlo-toolkit/package.nix:40]: I don't quite understand this. If "*-config" utilities come from "depsBuildHost", then `gfortran` should be here too. But then how strictDeps doesn't prevent gfortran from being found, but "*-config" utilities need this?

**Review by veprbl — APPROVED** [review]
LGTM

File: pkgs/by-name/fa/fastnlo-toolkit/package.nix (hunk 1 above)
Comment (FliegendeWurst) [reply pkgs/by-name/fa/fastnlo-toolkit/package.nix:40]: Simple, it seemingly never used gfortran:

> No Fortran extension requested. Continue without fortranext.

I can remove the gfortran bits or try to enable the extension, whichever you prefer.

Comment (veprbl) [reply pkgs/by-name/fa/fastnlo-toolkit/package.nix:40]: Maybe there was a silent regression at some point. However, I'm almost sure the extension is not useful. The following change worked for me:
```diff
diff --git a/pkgs/by-name/fa/fastnlo-toolkit/package.nix b/pkgs/by-name/fa/fastnlo-toolkit/package.nix
index d43cc2698c1b..6349072ac70f 100644
--- a/pkgs/by-name/fa/fastnlo-toolkit/package.nix
+++ b/pkgs/by-name/fa/fastnlo-toolkit/package.nix
@@ -3,7 +3,6 @@
   stdenv,
   fetchurl,
   boost,
-  gfortran,
   lhapdf,
   ncurses,
   perl,
@@ -38,8 +37,6 @@ stdenv.mkDerivation rec {
   buildInputs =
     [
       boost
-      gfortran
-      gfortran.cc.lib
       lhapdf
       yoda
     ]
```

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 366449` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python311Packages.fastnlo-toolkit</li>
    <li>python312Packages.fastnlo-toolkit</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>apfel</li>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>fastnlo-toolkit</li>
    <li>herwig</li>
    <li>lhapdf (python312Packages.lhapdf)</li>
    <li>mcfm</li>
    <li>nnpdf (python312Packages.nnpdf)</li>
    <li>professor</li>
    <li>pythia</li>
    <li>python311Packages.lhapdf</li>
    <li>python311Packages.nnpdf</li>
    <li>python311Packages.rivet</li>
    <li>python311Packages.yoda</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>yoda (python312Packages.yoda)</li>
    <li>sacrifice</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Need to move around more inputs to make python bindings build:
```diff
diff --git a/pkgs/by-name/fa/fastnlo-toolkit/package.nix b/pkgs/by-name/fa/fastnlo-toolkit/package.nix
index d43cc2698c1b..db79f1be9f83 100644
--- a/pkgs/by-name/fa/fastnlo-toolkit/package.nix
+++ b/pkgs/by-name/fa/fastnlo-toolkit/package.nix
@@ -3,7 +3,6 @@
   stdenv,
   fetchurl,
   boost,
-  gfortran,
   lhapdf,
   ncurses,
   perl,
@@ -35,25 +34,18 @@ stdenv.mkDerivation rec {
     ./yoda2_support.patch
   ];
 
-  buildInputs =
-    [
-      boost
-      gfortran
-      gfortran.cc.lib
-      lhapdf
-      yoda
-    ]
-    ++ lib.optional withPython python
-    ++ lib.optional (withPython && python.isPy3k) ncurses;
+  nativeBuildInputs = [
+    lhapdf
+    yoda
+  ] ++ lib.optional withPython python;
+  buildInputs = [
+    boost
+    lhapdf
+    yoda
+  ] ++ lib.optional withPython python ++ lib.optional (withPython && python.isPy3k) ncurses;
 
-  propagatedBuildInputs =
-    [
-      zlib
-    ]
-    ++ lib.optional withPython [
-      swig
-      python.pkgs.distutils
-    ];
+  propagatedNativeBuildInputs = lib.optional withPython [ swig ];
+  propagatedBuildInputs = [ zlib ] ++ lib.optional withPython [ python.pkgs.distutils ];
 
   preConfigure = ''
     substituteInPlace ./fastnlotoolkit/Makefile.in \
@@ -64,9 +56,7 @@ stdenv.mkDerivation rec {
     chmod +x check/fnlo-tk-stattest.pl.in
   '';
 
-  configureFlags = [
-    "--with-yoda=${yoda}"
-  ] ++ lib.optional withPython "--enable-pyext";
+  configureFlags = [ "--with-yoda=${yoda}" ] ++ lib.optional withPython "--enable-pyext";
 
   enableParallelBuilding = true;
 
@@ -84,6 +74,8 @@ stdenv.mkDerivation rec {
   # Use a workaround from https://github.com/swig/swig/issues/1538
   env.CXXFLAGS = "-D_LIBCPP_ENABLE_CXX17_REMOVED_FEATURES";
 
+  strictDeps = true;
+
   meta = with lib; {
     homepage = "http://fastnlo.hepforge.org";
     description = "Fast pQCD calculations for hadron-induced processes";
```

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 366449` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>apfel</li>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>fastnlo-toolkit</li>
    <li>herwig</li>
    <li>lhapdf (python312Packages.lhapdf)</li>
    <li>mcfm</li>
    <li>nnpdf (python312Packages.nnpdf)</li>
    <li>professor</li>
    <li>pythia</li>
    <li>python311Packages.fastnlo-toolkit</li>
    <li>python311Packages.lhapdf</li>
    <li>python311Packages.nnpdf</li>
    <li>python311Packages.rivet</li>
    <li>python311Packages.yoda</li>
    <li>python312Packages.fastnlo-toolkit</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>yoda (python312Packages.yoda)</li>
    <li>sacrifice</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Pull request merged**